auth/basic_auth.go

```diff
@@ -15,12 +15,39 @@ func DecorateWithBasicAuth(next http.HandlerFunc, credentials *BasicAuthCredenti
 		w.Header().Set("WWW-Authenticate", `Basic realm="Restricted"`)
 
 		if !ok || !(credentials.Password == password && user == credentials.User) {
+			unauthorized(w, r)
+		}
+
+		next.ServeHTTP(w, r)
+	}
+}
+
+// Support for Multiple User Basic Auth Providers
+func DecorateWithMultiUserBasicAuth(next http.HandlerFunc, creds []BasicAuthCredentials) http.HandlerFunc {
```

Comment:
Is is possible to use a `map[string]BasicAuthCredentials` here? this would ensure a constant time lookup and that a user has exactly one password.  With the slice, it is possible for a user to be repeated with different passwords, unless that is a desirable behavior.